org.jacoco.report/src/org/jacoco/report/check/Limit.java

```diff
@@ -186,4 +195,29 @@ private String message(final String minmax, final BigDecimal v,
 				rounded.toPlainString(), minmax, ref.toPlainString());
 	}
 
+	private String checkRatioLimit() {
+		if (CounterValue.MISSEDRATIO.equals(value)
+				|| CounterValue.COVEREDRATIO.equals(value)) {
+			final String minmsg = checkRatioLimit("minimum", minimum);
+			if (minmsg != null) {
+				return minmsg;
+			}
+			final String maxmsg = checkRatioLimit("maximum", maximum);
+			if (maxmsg != null) {
+				return maxmsg;
+			}
+		}
+		return null;
+	}
+
+	private String checkRatioLimit(final String minmax, final BigDecimal v) {
+		if (v != null && (v.compareTo(BigDecimal.ZERO) == -1
+				|| v.compareTo(BigDecimal.ONE) == 1)) {
```

Comment:
While `BigDecimal.compareTo` clearly specifies return values as `-1, 0, or 1`,
`< 0` (less than zero) and `> 0` (greater than 0) look clearer to my taste,
also `Comparable.compareTo` specifies return values as `a negative integer, zero, or a positive`.